test/testcafe/spec/TerminalView_spec.js

```diff
@@ -195,44 +195,84 @@ test.requestHooks(terminalCustomAccessDeniedErrorMessageMock)('should render cus
   await t.expect(terminalViewPage.form.getErrorBoxHtml()).eql('<span data-se="icon" class="icon error-16"></span><div class="custom-access-denied-error-message"><p>You do not have permission to perform the requested action.</p><ul class="custom-links"><li><a href="https://www.okta.com/" target="_blank" rel="noopener noreferrer">Help link 1</a></li><li><a href="https://www.okta.com/help?page=1" target="_blank" rel="noopener noreferrer">Help link 2</a></li></ul></div>');
 });
 
+const assertAnchorWithBlankTargetHaveCorrectRelAttribute = async (t, selector) => {
+  await t.expect(
+    selector
+      .find('a[target="_blank"]')
+      .filter((node) => {
+        const relValues = (node.getAttribute('rel') || '').split(' ');
+        return relValues.indexOf('noopener') < 0 || relValues.indexOf('noreferrer') < 0;
+      })
+      .exists).eql(false);
+};
+
+const assertSIWEndUserRemediationHeaders = async (t, terminalViewPage, explanationExists) => {
+  await t.expect(terminalViewPage.form.getErrorBox().find('.end-user-remediation-title').withExactText('Your device doesn\'t meet the security requirements').exists).eql(true);
+  await t.expect(terminalViewPage.form.getErrorBox().find('.end-user-remediation-explanation').withExactText('To sign in, make the following updates. Then, access the app again.').exists).eql(explanationExists);
```

Comment:
We want to avoid selecting (and particularly sub-selecting) on exact classnames in tests. We're undergoing an effort to eliminate all selectors in spec files, and try to eliminate whenever possible CSS classname selectors in page objects / form objects as well.

Would asserting that `terminalViewPage.form.getErrorBoxText().contains('target string').eql(true)` in these be sufficient?

Reply:
I have changed some of the tests to not be as specific with selecting elements. Let me know if there are any more changes I should make.